pkg/dynamic/client.go

```diff
@@ -107,12 +113,17 @@ func (c *knDynamicClient) ListSources(types ...WithType) (*unstructured.Unstruct
 	var (
 		sourceList               unstructured.UnstructuredList
 		options                  metav1.ListOptions
-		numberOfsourceTypesFound int
+		numberOfSourceTypesFound int
 	)
 	sourceTypes, err := c.ListSourcesTypes()
 	if err != nil {
 		return nil, err
 	}
+
+	if sourceTypes == nil || len(sourceTypes.Items) == 0 {
```

Comment:
I can comment this condition (both parts around ||) and still pass all tests...

Reply:
added tests